File: ruby/user_management_policy/lib/user_management/create_user.rb
require 'securerandom'
require_relative 'user_id'
require_relative 'user'
require_relative 'user_repository'

module UserManagement
  class CreateUser
    def initialize(user_repository)
      @user_repository = user_repository
    end

    def call(email, display_name)
      user_id = UserId.new(value: SecureRandom.uuid)
      @user_repository.save(User.new(id: user_id, email: email, display_name: display_name))
      user_id
    end
  end
end

File: python/user_management_policy/create_user.py
import uuid
from user_management_policy.user import User
from user_management_policy.user_id import UserId
from user_management_policy.user_repository import UserRepository


def make_create_user(user_repository: UserRepository):
    def create_user(email: str, display_name: str) -> UserId:
        user_id = UserId(value=str(uuid.uuid4()))
        user_repository.save(User(id=user_id, email=email, display_name=display_name))
        return user_id

    return create_user
